vendor/github.com/keybase/gomounts/darwin.go

```diff
@@ -0,0 +1,47 @@
+// +build darwin
+
+package gomounts
+
+/*
+#include <sys/param.h>
+#include <sys/ucred.h>
+#include <sys/mount.h>
+*/
+import "C"
+import (
+	"errors"
+	"reflect"
+	"strconv"
+	"sync"
+	"unsafe"
+)
+
+var mtx sync.Mutex = sync.Mutex{}
+
+func getMountedVolumes() ([]Volume, error) {
+	// getmntinfo is non-reentrant
+	mtx.Lock()
+	defer mtx.Unlock()
+
+	result := make([]Volume, 0)
+
+	var mntbuf *C.struct_statfs
+	count := int(C.getmntinfo(&mntbuf, C.MNT_NOWAIT))
```

Comment:
this code is buggy:

- `getmntinfo` return *0* on an error and sets `errno`.
- would feel much better using the thread-safe version `getmntinfo_r_np` that doesn't shrae a buffer.
- the code casting the array to a slice below is kinda scary -- https://golang.org/pkg/reflect/#SliceHeader says `Data` might still be garbage collected? would much rather just iterate manually somehow.